[Lim Boon Kee] iP

bklimey opened the pull request:
# Welcome to Duke Point O!

> "why waste you brain space for boring stuff" -nobody

Duke Point O helps you keep track of mundane tasks so that you can free your brain
~~and let them rot~~ for greater stuff!

Who doesn't like task list??
Everybody needs one!
- [ ] Do me
- [ ] Do me!
- [ ] Do me!!

### What it can do
- Add task
- Mark task as done
- Delete task
- Okay I know it sounds boring already
- but I assure you it is **AWESOME**!!1!
- And many more...

### Try it (at your own risk)!
1. Go to my recent [release](https://github.com/bklimey/ip/releases/tag/A-Jar) and download `ip.jar`
1. Open command prompt in the same folder as `ip.jar` and enter `java -jar ip.jar` into the command prompt
1. Run it and have fun!

*Disclaimer: Not responsible for any mishaps that happens to your computer*
   
### END

Btw, I liek Python :snake:
```python
def isAwesome():
  return True
```

(Note: The above write-up is for trying out formatting, please do not take it seriously)

Commit:
Add Gradle support

bklimey committed:
Level-1

bklimey committed:
Level-2

bklimey committed:
Level-3

bklimey committed:
Level-4

bklimey committed:
A-TextUiTesting

bklimey committed:
Level-5

bklimey committed:
cleaning up of code

bklimey committed:
Level-6

bklimey committed:
Reorganize task list into a class

bklimey committed:
A-Enums

bklimey committed:
Level-7

bklimey committed:
Level-7

bklimey committed:
Level-8

bklimey committed:
A-MoreOOP

bklimey committed:
A-Packages

lwlshawn reviewed (commented):
Your code looks good! Only major thing missing from the course coding standard right now is javadoc comments I think.

bklimey committed:
A-JUnit

bklimey committed:
Merge branch 'branch-Level-8'

bklimey committed:
Refactor package

bklimey committed:
A-JavaDoc

bklimey committed:
A-CodingStandard

bklimey committed:
Level-9

bklimey committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/TaskList.java

bklimey committed:
Merge branch 'branch-A-CodingStandard'

bklimey committed:
Change README.md

bklimey committed:
Fix formatting

bklimey committed:
Update README.md

bklimey committed:
Update README.md

bklimey committed:
Update README.md

NicolasCwy commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+        this.by = by;
+    }
+
+    @Override
```
Could consider adding javadocs for functions like these!

NicolasCwy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,91 @@
+    }
+
+    private static Task convertStringToTask(String taskString) throws IllegalArgumentException {
+        // Example format: T | 1 | read book
```
Spotted a few places like this which could do with javadocs!

NicolasCwy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,91 @@
+
+    private static final String FILE_PATH = "./data/duke.txt";
+
+    private static final Set<String> VALID_TASK_TYPE = Set.of("T", "D", "E");
```
Nice use of casing for constants! Can consider using the plural VALID_TASK_TYPES, since it is a set.
Same as VALID_DONE_STATUSES

NicolasCwy commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
I like the naming of boolean attribute 👍

NicolasCwy reviewed (commented):
Congratulations on your first PR!🎉 Overall your code follows the coding standards for variable names. You could consider adding javadocs as @lwlshawn suggested, throughout your code or if you have less time on more cryptic code methods such as `serialise`.

NicolasCwy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,91 @@
+    private static final String FILE_PATH = "./data/duke.txt";
+
+    private static final Set<String> VALID_TASK_TYPE = Set.of("T", "D", "E");
+    private static final Set<String> VALID_DONE_STATUS = Set.of("0", "1");
```
Nice use of casing for constants! Can consider using the plural VALID_TASK_TYPES and VALID_DONE_STATUSES, since they are a set of values.

noobmaster19 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+    public Deadline(String description, boolean isDone, String by) {
+        super(description, isDone);
+        this.by = by;
+    }
```
Yes please do consider adding javadocs

noobmaster19 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,91 @@
+        scanner.close();
+        return taskList;
+    }
+
```
The logic here seems quite thick. Might want to explain what you are doing here

noobmaster19 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,91 @@
+import java.util.Set;
+import java.util.List;
+
+/**
```
Great job not using the wildcard for imports!

noobmaster19 commented on src/main/java/duke/TimeHandler.java:
```diff
@@ -0,0 +1,42 @@
+public class TimeHandler {
+
+    /**
+     * Returns a formatted data and time string.
```
Great helper function

noobmaster19 replied:
i mean class!

noobmaster19 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,76 @@
+                int number = Integer.parseInt(inp.substring(5));
+                userList.markDone(number);
+            } else if (inp.startsWith("delete ")) { //check if command is "delete"
+                int number = Integer.parseInt(inp.substring(7));
```
might want to abstract this into the parser

bklimey committed:
Add code that was accidentally deleted

bklimey committed:
Merge branch 'add-gradle-support'

bklimey committed:
Refactor code to adhere to coding standard

lwlshawn commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,54 @@
+package duke;
+
+public enum Command {
+    DONE {
+        public String toString() {
+            return "done";
```
Just a small nit but including documentation about how the class is meant to be used could be helpful for other developers (in theory). Sorry for commenting on this again, 2103 required review comments and not general ones.

lwlshawn reviewed (commented):
for 2103 script

bklimey committed:
Level-10

bklimey committed:
Level-10

bklimey committed:
A-Gradle

bklimey committed:
Merge branch 'branch-Level-10'

noobmaster19 reviewed (commented):
Review

bklimey committed:
Remove GUI code from Duke class

Duke class is not longer managing the GUI.

Let's remove code that is no longer relevant in Duke class to avoid confusion and improve code readability.

bklimey committed:
A-Assertions

bklimey committed:
Merge pull request #2 from bklimey/branch-A-Assertions

Add assertions to TaskList class

bklimey committed:
A-CodeQuality

bklimey committed:
Merge pull request #3 from bklimey/branch-A-CodeQuality

A-CodeQuality

bklimey committed:
BCD-Extension

bklimey committed:
Set theme jekyll-theme-merlot

bklimey committed:
A-BetterGui

bklimey committed:
Merge remote-tracking branch 'origin/master'

bklimey committed:
Remove Ui class

Ui was previously used for text-based user interaction. Since the program has pivoted to GUI-based, Ui class is no longer relevant.

To avoid confusion, let's remove the Ui class.

bklimey committed:
A-UserGuide

bklimey committed:
BCD-Extension for C-Undo

bklimey committed:
Handle find command with no input

bklimey committed:
Fix parseTime to correctly convert 0000

Previously, 0000 would convert to 0:00 AM. Now, it would convert to 12:00 AM.

bklimey committed:
Add test for TimeHandler class

bklimey committed:
Refactor code to increase readability

bklimey committed:
Add support for undo

bklimey committed:
Update GUI

bklimey committed:
Update README.md

bklimey committed:
Update README.md
